src/main/java/org/example/workerapi/service/EmployeeService.java

```diff
@@ -78,4 +85,46 @@ public void addTestEmployees(){
         }
     }
 
+    @Transactional
+    public void deleteEmployeeByID(Long id) {
+        Optional<Employee> maybeEmployeeFromDb = employeeRepository.findById(id);
+        if (maybeEmployeeFromDb.isEmpty()) {
+            throw new EntityNotFoundException("Employee with id " + id + " not found");
+        }
+        employeeRepository.deleteById(id);
+    }
+
+    public Employee patchEmployeeById(Long id, Employee employeeFromRequest) {
+        Optional<Employee> maybeEmployeeFromDb = employeeRepository.findById(id);
+        if (maybeEmployeeFromDb.isEmpty()) {
+            throw new EntityNotFoundException("Employee with id " + id + " not found");
```

Comment:
Tvarkingai handlini errorus jei kažkas buvo nerasta ar pan. Pabandyk panaudoti ControllerAdvice, kad tvarkingai suvaldyti HTTP error kodus:)

https://www.baeldung.com/exception-handling-for-rest-with-spring#controlleradvice